src/smt.rs

```diff
@@ -1005,7 +1005,7 @@ impl SparseMerkleTree {
         Ok(new_subtree)
     }
 
-    fn update_one_node(&self, info: &UpdateNodeInfo) -> Result<(Vec<Node>, Vec<u8>), SMTError> {
+    fn update_one_node(&self, info: &UpdateNodeInfo) -> Result<Option<(Node, u8)>, SMTError> {
```

Comment:
It would be better to return `StructurePosition` strong type instead of `u8`

Reply:
Done.